### PR TITLE
Added help tabs and nonces

### PR DESCRIPTION
**Pull request by mnelson4**
## Problem this Pull Request solves

Added help tabs (one for most steps) and use nonces in admin pages, and slight reorganization of attendee importer pages.

## How has this been tested

* [x] Visit the attendee importer. Notice the help tabs, and if you click on the question mark next to the current step's name, the corresponding help tab will be opened
* [x] clicking on the "Import" tab at the top doesn't produce any error, and the help tabs still work afterwards, and you can still import attendees afterwards too

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.

### CODE REVIEW

File: src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php [hunk 1]
```diff
@@ -99,6 +102,24 @@ protected function getFormStepsCollection()
...
+     * @throws InvalidIdentifierException
+     * @throws InvalidInterfaceException
+     * @throws InvalidFormHandlerException
+     * @throws InvalidArgumentException
+     */
+    public function getSteps()
+    {
+        $this->getFormStepsCollection();
+        // Don't return the original steps collection as when we loop through it, it borks later logic.
+        return clone $this->form_steps;
```
Comment (tn3rb) [inline src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php]: > Don't return the original steps collection as when we loop through it, it borks later logic.

how is it borking the logic?
is it a case where you break out of a loop and then the next loop doesn't start at the beginning of the collection?
if so, you can issue a `rewind()` command on the collection to start from the beginning.

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<p><?php esc_html_e('Before Using the Event Espresso 4 Attendee Importer', 'event_espresso'); ?></p>
+<p>
+    <strong><?php esc_html_e('Backup', 'event_espresso');?></strong>
+    <?php esc_html_e("Always make sure you have a complete database backup. If you import data improperly, or there is an error, this is the best way to undo the import.", 'event_espresso'); ?>
+</p>
+<p>
+    <strong><?php esc_html_e('Setup The Event in Event Espresso', 'event_espresso');?></strong>
+    <?php esc_html_e("You must first create the event and ticket you wish to import the attendees to. Currently, only importing to a single event's ticket is supported.", 'event_espresso'); ?>
```
Comment (tn3rb) [inline src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: > Currently, only importing to a single event's ticket is supported.

hmmm... not completely clear... maybe something like:

```
Currently, only importing to one event ticket at a time is supported.
```

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+</p>
+<p>
+    <strong><?php esc_html_e('Prepare CSV File', 'event_espresso');?></strong>
+    <?php esc_html_e("Create your Comma-Separated-Value (CSV) file. The first row should be column headers, and each row subsequent represents a registration.", 'event_espresso'); ?>
```
Comment (tn3rb) [inline src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: > and each row subsequent represents a registration.

maybe:

```
and each subsequent row represents a registration.
```

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+<p>
+    <?php esc_html_e("You can give any name you want to the column headers. But there must be a column for:", 'event_espresso'); ?>
+</p>
+<ul>
+    <li><?php esc_html_e('First Name', 'event_espresso');?></li>
+    <li><?php esc_html_e('Email', 'event_espresso');?></li>
+</ul>
+<p><?php esc_html_e('You can also add columns for any of the following:', 'event_espresso');?></p>
+<ul>
+    <li><?php esc_html_e('Any other system questions, like Last Name, Address, City, State, Country, Zip, or Phone. (Please note: for state you can use the state’s name, abbreviation, or database ID; for country you can use its name or ISO2 or ISO3 code)', 'event_espresso'); ?></li>
```
Comment (tn3rb) [inline src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: > for country you can use its name or ISO2 or ISO3 code

maybe something like:

- for country you can use its name, ISO2 code, or ISO3 code
- for country you can use its name or ISO country code (ISO2 or ISO3)

or maybe even add examples in:

- for country you can use its name, ISO2 code (ex: US), or ISO3 code (ex: USA)

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <li><?php esc_html_e('Any custom questions. The column header’s text does not need to match the custom question. For checkbox or multi-select questions, separate the different answer values with a pipe "|". Eg "', 'event_espresso'); ?></li>
+    <li><?php esc_html_e('The payment amount, if the registrations are for a paid ticket.', 'event_espresso'); ?></li>
```
Comment (tn3rb) [inline src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: > For checkbox or multi-select questions, separate the different answer values with a pipe "|". Eg "
> The payment amount, if the registrations are for a paid ticket.

??? is the example missing?

File: src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php (hunk 1 above)
Comment (mnelson4) [reply src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php]: > is it a case where you break out of a loop and then the next loop doesn't start at the beginning of the collection?
Yes, exactly. This was it:
error if no clone


> if so, you can issue a rewind() command on the collection to start from the beginning.

I did that (added `$steps->rewind()` in `Attendee_Importer_Admin_Page::_set_page_config()` just before `$this->_page_config = array( ...`) , and it seemed to fix it, but then it always displayed the first step's form (even though it does highlight the correct progress step). Like so:
logic error if reset

You should be able to reproduce if you like.

Does that give you any clues why looping through the steps earlier on breaks things?

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php (hunk 2 above)
Comment (mnelson4) [reply src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: :+1: thanks

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php (hunk 3 above)
Comment (mnelson4) [reply src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: thanks

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php (hunk 4 above)
Comment (mnelson4) [reply src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: yeah good idea to give some examples

File: src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php (hunk 5 above)
Comment (mnelson4) [reply src/ui/admin/attendee_importer/help_tabs/attendee_importer_import_overview.help_tab.php]: Why yes it is!

**Comment by mnelson4** [comment]
Implemented your suggested phrasing tweaks, made a few more tweaks that seemed good to me, and responded with some more info on https://github.com/eventespresso/eea-attendee-importer/pull/33#discussion_r278674875

File: src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php (hunk 1 above)
Comment (tn3rb) [reply src/domain/services/import/csv/attendees/forms/form_handlers/StepsManager.php]: pretty sure all that is happening is the pointer for the current item in the collection is off. Probably just need to rewind and then advance to the step that matches the current slug. I'll checkout your branch and have a look.

**Comment by tn3rb** [comment]
so what was happening is the `SequentialStepFormManager` was building the form and setting the current step prior to your looping through the steps in `Attendee_Importer_Admin_Page::_set_page_config()` to build the help tabs.
Just had to call `SequentialStepFormManager::setCurrentStepFromRequest()` again after that in order to advance back to the correct step.
That method is protected so this required overriding it in `\EventEspresso\AttendeeImporter\domain\services\import\csv\attendees\forms\form_handlers\StepsManager` to make it public.

**Comment by joshfeck** [comment]
This was tested.

**Pull request merged**